File: backend/backend/Models/Story.cs
﻿using System.Text.Json.Serialization;

namespace backend.Models
{
    public class Story
    {
        public int StoryId { get; set; }
        public string Title { get; set; } = string.Empty; // Ensure non-nullable
        public string? Description { get; set; }
        public string? Content { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? ImageUrl { get; set; }
        public int? StoryGroupId { get; set; }

        public ICollection<Character> Characters { get; set; } = new List<Character>();
        [JsonIgnore]
        public StoryGroup StoryGroup { get; set; }
        [JsonIgnore]
        public ICollection<StoryPart> StoryParts { get; set; } = new List<StoryPart>();

        
    }
}

File: frontend/src/models/Story.ts
import { User } from './User';
import { StoryPart } from './StoryPart';
import { Character } from './Character';

export interface Story {
  storyId: number;
  title: string;
  description: string;
  content: string | null;
  createdAt: string;
  userId: number;
  imageUrl: string;
  user: User | null;
  characters: Character[];
  storyParts: StoryPart[];
}
